# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1336007680 14400
#      Wed May 02 21:14:40 2012 -0400
# Node ID fd6e50e2eee6499df77eb16572b4ab3417da2fc9
# Parent  c38a253723d31d6a7611ebd7616bca59017ffeed
legend() should accept labels being specified as both cellstrs and char.

legend.m: Accept labels as both cell strings and character strings.

diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -267,16 +267,19 @@ function [hlegend2, hobjects2, hplot2, t
         nargs = numel (varargin);
       endif
     elseif (iscellstr (arg))
       varargin = arg;
       nargs = numel (varargin);
     else
       error ("legend: expecting argument to be a character string");
     endif
+  elseif (nargs > 1 && iscellstr (varargin{1}))
+    varargin = {varargin{1}{:}, varargin{2:end}};
+    nargs = numel (varargin);
   endif
 
   if (strcmp (show, "off"))
     if (! isempty (hlegend))
       set (findobj (hlegend), "visible", "off");
       hlegend = [];
     endif
     hobjects = [];
@@ -964,16 +967,21 @@ function updateline (h, d, hlegend, line
             "markerfacecolor", get (h, "markerfacecolor"),
             "markersize", get (h, "markersize"), "linestyle", "none",
             "userdata", h, "parent", hlegend);
     endif
   endif
 endfunction
 
 %!demo
+%! plot (rand (2))
+%! legend ({'foo'}, 'bar', 'boxoff')
+%! title ('legend() should warn about an extra label')
+
+%!demo
 %! plot (rand (2,2)) ;
 %! h = legend ('a', 'b') ;
 %! legend ('right') ;
 %! set (h, 'textposition', 'left')
 %! set (h, 'textposition', 'right')
 %! set (h, 'textcolor', [1 0 1])
 
 %!demo
